packages/abi-typegen/test/fixtures/templates/contract/factory.hbs.ts

```diff
@@ -1,4 +1,4 @@
-/* Autogenerated file. Do not edit manually. */
+export default `/* Autogenerated file. Do not edit manually. */
```

Comment:
There are these `export default`s before the `Autogenerated file` notice in all of these files. Is this expected behaviour?

Reply:
TL;DR: Yes, it is.

These files used to be Handlebars templates (`*.hbs`), but now they've become plain TS (`*.hbs.ts`), that export a default `string`. This `string` is a **_Handlebars_** template. The change is subtle; it can go unnoticed.

This commit says it all: https://github.com/FuelLabs/fuels-ts/pull/827/commits/16679ddc36090d009d0d9810042016146e7e6f21


----

This was one of the side effects of dropping `tsup`, which was responsible for bundling the `.hbs` files as `string` using a text-loader. We're using `tsc` directly now, which can't do bundling, so another approach was required.

Instead of adding _yet another_ pre-build step in the mix (_which I did already a few PRs back_), I preferred to provide plain Typescript files, simplify everything and move on. These files export a `string`, and this `string` is a **_Handlebars_** template. All the rest is the same; I just jumped over the bundling step.

Handlebars syntax coloring in VSCode is tricked via file associations:

![assoc](https://user-images.githubusercontent.com/26660/225718906-704352b4-4222-4ba6-8886-5c881e4a9d43.png)